conan/api/subapi/report.py

```diff
@@ -0,0 +1,171 @@
+
+        old_diff_path = os.path.abspath(os.path.join(old_cache_path, os.path.pardir)).replace("\\",
+                                                                                              "/")
+        new_diff_path = os.path.abspath(os.path.join(new_cache_path, os.path.pardir)).replace("\\",
+                                                                                              "/")
+
+        src_prefix = base64.b64encode(str(new_export_ref).encode()).decode() + "/"
+        dst_prefix = base64.b64encode(str(old_export_ref).encode()).decode() + "/"
+
+        command = (f'git diff --no-index --src-prefix {src_prefix} --dst-prefix {dst_prefix} '
+                   f'"{old_diff_path}" "{new_diff_path}"')
+
+        ConanOutput().info(
+            f"Generating diff from {old_export_ref.repr_notime()} to {new_export_ref.repr_notime()} (this might take a while)")
+        ConanOutput().info(command)
+
+        # TODO: This is internal, we should use the public API, but nothing exposes functionality like this
+        # We ignore the errors because git diff returns 1 if there are differences
+        diff = _execute_command(command, ignore_error=True)
+
+        if old_path:
+            self.conan_api.remove.recipe(old_export_ref)
+        if new_path:
+            self.conan_api.remove.recipe(new_export_ref)
+
+        return {
+            "diff": diff,
+            "old_export_ref": old_export_ref,
+            "new_export_ref": new_export_ref,
+            "old_diff_path": old_diff_path,
+            "new_diff_path": new_diff_path,
+            "src_prefix": src_prefix,
+            "dst_prefix": dst_prefix,
+        }
+
+def _configure_source(conan_api, conanfile_path, ref, remotes):
+    app = ConanApp(conan_api)
+    conanfile = app.loader.load_consumer(conanfile_path, name=ref.name, version=str(ref.version),
+                                         user=ref.user, channel=ref.channel, graph_lock=None,
+                                         remotes=remotes)
+    # This profile is empty, but with the conf from global.conf
+    profile = conan_api.profiles.get_profile([])
+    initialize_conanfile_profile(conanfile, profile, profile, CONTEXT_HOST, False)
+    # This is important, otherwise the ``conan source`` doesn't define layout and fails
+    if hasattr(conanfile, "layout"):
+        with conanfile_exception_formatter(conanfile, "layout"):
+            conanfile.layout()
+
+    recipe_layout = app.cache.recipe_layout(ref)
+    export_source_folder = recipe_layout.export_sources()
+    source_folder = recipe_layout.source()
+
+    conanfile.folders.set_base_source(source_folder)
+    conanfile.folders.set_base_export_sources(export_source_folder)
+    conanfile.folders.set_base_recipe_metadata(recipe_layout.metadata())
+    config_source(export_source_folder, conanfile, conan_api.config.hook_manager)
+
+def _execute_command(cmd, stderr=None, ignore_error=False):
```

Comment:
This is mostly the same as `check_runner_output` but ignores encoding errors, maybe we can add a flag to that funcion and use it instead